stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -1515,7 +1515,7 @@ end
 end
 
 # apply reflector from left
-@inline function reflectorApply!(x::AbstractVector, τ::Number, A::AbstractMatrix)
+@inline function reflectorApply!(x::AbstractVector, τ::Number, A::AbstractVecOrMat)
     require_one_based_indexing(x)
     m, n = size(A)
```

Comment:
You'll need to rely on implicit dimension extension (or however that is called), otherwise the destructuring will fail for vectors:
```suggestion
    m, n = size(A, 1), size(A, 2)
```